src/lib/CMakeLists.txt

```diff
@@ -32,40 +32,33 @@ set(SOURCES access.cpp
             SoftHSM.cpp
             )
 
-set(DEPENDENCIES softhsm_common-obj
-                 softhsm_crypto-obj
-                 softhsm_datamgr-obj
-                 softhsm_handlemgr-obj
-                 softhsm_objectstore-obj
-                 softhsm_sessionmgr-obj
-                 softhsm_slotmgr-obj
-                 )
-
-set(OBJECT_FILES $<TARGET_OBJECTS:softhsm_common-obj>
-                 $<TARGET_OBJECTS:softhsm_crypto-obj>
-                 $<TARGET_OBJECTS:softhsm_datamgr-obj>
-                 $<TARGET_OBJECTS:softhsm_handlemgr-obj>
-                 $<TARGET_OBJECTS:softhsm_objectstore-obj>
-                 $<TARGET_OBJECTS:softhsm_sessionmgr-obj>
-                 $<TARGET_OBJECTS:softhsm_slotmgr-obj>
+set(STATIC_FILES softhsm_common
+                 softhsm_crypto
+                 softhsm_datamgr
+                 softhsm_handlemgr
+                 softhsm_objectstore
+                 softhsm_sessionmgr
+                 softhsm_slotmgr
                  )
 
 
 ###############################################################################
 # Static Library Config
 ###############################################################################
-add_library(${PROJECT_NAME}-static STATIC ${SOURCES} ${OBJECT_FILES})
-add_dependencies(${PROJECT_NAME}-static ${DEPENDENCIES})
-set_target_properties(${PROJECT_NAME}-static
-                      PROPERTIES OUTPUT_NAME ${PROJECT_NAME}
-                      )
-generate_export_header(${PROJECT_NAME}-static)
+if(ENABLE_STATIC)
+        add_library(${PROJECT_NAME}-static STATIC ${SOURCES})
+        target_link_libraries(${PROJECT_NAME}-static ${STATIC_FILES})
+        set_target_properties(${PROJECT_NAME}-static
+                              PROPERTIES OUTPUT_NAME ${PROJECT_NAME}
+                              )
+        generate_export_header(${PROJECT_NAME}-static)
+endif()
 
 ###############################################################################
 # Shared Library Config
 ###############################################################################
-add_library(${PROJECT_NAME} SHARED ${SOURCES} ${OBJECT_FILES})
```

Comment:
To fix linker problem, why not add:
```
target_link_libraries(${PROJECT_NAME} ${OPENSSL_LIBRARIES})
```

Reply:
Nevermind, I like your suggestion better. 

Place the dependency on libsofthsm_crypto and then propagate up.